Upgrade to Scalameta 3.0.0

xeno-by opened the pull request.

xeno-by committed:
Upgrade to Scalameta 3.0.0

olafurpg commented on sbthost/nsc/src/main/scala/scala/meta/internal/sbthost/RangeOps.scala:
```diff
@@ -0,0 +1,35 @@
+    private def line(offset: Int): Int = {
+      // NOTE: Looks like some seriously crazy stuff is going on here.
+      // source.offsetToLine(offset)
+      0
```
I expanded on "crazy stuff" and included a comment explaining why this hack works for scala.meta.Position but is incompatible with how LSP treats line/character pairs.

olafurpg replied:
In LSP, characters=Int.MaxValue falls back to the last character in the line https://microsoft.github.io/language-server-protocol/specification#position

olafurpg committed:
Document line/character hack.

The build did not trigger recompilation of test inputs when changing the
compiler plugin.

olafurpg reviewed (approved):
LGTM 👍
